File: RepoTransBench/target_projects/Java/C#/tamir7_Contacts/public_tests/EventPublicTest.cs
using Xunit;
using Contacts;

namespace Contacts.PublicTests
{
    public class EventPublicTest
    {
        [Fact]
        public void TestConstructorsAndGetters_Type()
        {
            var e = new Event("2050-12-31", Event.Type.ANNIVERSARY);
            Assert.Equal("2050-12-31", e.StartDate);
            Assert.Equal(Event.Type.ANNIVERSARY, e.TypeValue);
            Assert.Null(e.Label);
        }

        [Fact]
        public void TestConstructorsAndGetters_Label()
        {
            var e = new Event("2024-07-14", "Special Date");
            Assert.Equal("2024-07-14", e.StartDate);
            Assert.Equal(Event.Type.CUSTOM, e.TypeValue);
            Assert.Equal("Special Date", e.Label);
        }

        [Fact]
        public void TestEqualsAndHashCode()
        {
            var e1 = new Event("2000-01-01", Event.Type.OTHER);
            var e2 = new Event("2000-01-01", Event.Type.OTHER);
            var e3 = new Event("2000-01-01", "Anniv");
            Assert.Equal(e1, e2);
            Assert.NotEqual(e1, e3);
            Assert.Equal(e1.GetHashCode(), e2.GetHashCode());
        }

        [Fact]
        public void TestTypeFromValue()
        {
            Assert.Equal(Event.Type.CUSTOM, Event.fromValue(-1));
            Assert.Equal(Event.Type.ANNIVERSARY, Event.fromValue(1));
            Assert.Equal(Event.Type.OTHER, Event.fromValue(2));
            Assert.Equal(Event.Type.BIRTHDAY, Event.fromValue(3));
            Assert.Equal(Event.Type.UNKNOWN, Event.fromValue(500));
        }

        [Fact]
        public void TestNotEqualConditions()
        {
            var e1 = new Event("2029-11-11", Event.Type.OTHER);
            Assert.False(e1.Equals(null));
            Assert.False(e1.Equals(new object()));
            var e2 = new Event("2222-02-22", Event.Type.OTHER);
            Assert.NotEqual(e1, e2);
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/tamir7_Contacts/public_tests/event_public_test.go
package public_tests

import (
	"testing"

	"github.com/tamir7/contacts"
)

func TestEventPublic_ConstructorsAndGetters_Type(t *testing.T) {
	e := contacts.NewEventWithType("2050-12-31", contacts.EventTypeANNIVERSARY)
	if e.StartDate != "2050-12-31" {
		t.Errorf("expected 2050-12-31, got %v", e.StartDate)
	}
	if e.Type != contacts.EventTypeANNIVERSARY {
		t.Errorf("expected ANNIVERSARY, got %v", e.Type)
	}
	if e.Label != "" {
		t.Errorf("expected empty label, got %v", e.Label)
	}
}

func TestEventPublic_ConstructorsAndGetters_Label(t *testing.T) {
	e := contacts.NewEventWithLabel("2024-07-14", "Special Date")
	if e.StartDate != "2024-07-14" {
		t.Errorf("expected 2024-07-14, got %v", e.StartDate)
	}
	if e.Type != contacts.EventTypeCUSTOM {
		t.Errorf("expected type CUSTOM, got %v", e.Type)
	}
	if e.Label != "Special Date" {
		t.Errorf("expected label 'Special Date', got %v", e.Label)
	}
}

func TestEventPublic_EqualsAndHashCode(t *testing.T) {
	e1 := contacts.NewEventWithType("2000-01-01", contacts.EventTypeOTHER)
	e2 := contacts.NewEventWithType("2000-01-01", contacts.EventTypeOTHER)
	e3 := contacts.NewEventWithLabel("2000-01-01", "Anniv")
	if !e1.Equal(e2) {
		t.Errorf("expected e1 == e2")
	}
	if e1.Equal(e3) {
		t.Errorf("expected e1 != e3")
	}
	if e1.HashCode() != e2.HashCode() {
		t.Errorf("expected hashcode match")
	}
}

func TestEventPublic_TypeFromValue(t *testing.T) {
	if contacts.EventTypeFromValue(-1) != contacts.EventTypeCUSTOM {
		t.Errorf("expected -1 -> CUSTOM")
	}
	if contacts.EventTypeFromValue(1) != contacts.EventTypeANNIVERSARY {
		t.Errorf("expected 1 -> ANNIVERSARY")
	}
	if contacts.EventTypeFromValue(2) != contacts.EventTypeOTHER {
		t.Errorf("expected 2 -> OTHER")
	}
	if contacts.EventTypeFromValue(3) != contacts.EventTypeBIRTHDAY {
		t.Errorf("expected 3 -> BIRTHDAY")
	}
	if contacts.EventTypeFromValue(500) != contacts.EventTypeUNKNOWN {
		t.Errorf("expected 500 -> UNKNOWN")
	}
}

func TestEventPublic_NotEqualConditions(t *testing.T) {
	e1 := contacts.NewEventWithType("2029-11-11", contacts.EventTypeOTHER)
	if e1.Equal(nil) {
		t.Errorf("should not equal nil")
	}
	if e1.Equal(struct{}{}) {
		t.Errorf("should not equal arbitrary type")
	}
	e2 := contacts.NewEventWithType("2222-02-22", contacts.EventTypeOTHER)
	if e1.Equal(e2) {
		t.Errorf("should not equal a different event")
	}
}